### PR TITLE
[fmt] Update to 11.0.0

### LINKED ISSUE #39660: [fmt] update to <11.0.0>
### Library name

fmt

### New version number

11.0.0

### Other information that may be useful (release notes, etc...)

https://github.com/fmtlib/fmt/releases/tag/11.0.0

### PR DESCRIPTION
**Pull request by waywardmonkeys**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by MonicaLiu0311** [comment]
Please get [failure logs here for x64-windows](https://dev.azure.com/vcpkg/public/_build/results?buildId=104559&view=logs&j=7922e5c4-0103-5f8f-ad17-45ce9bb98e80&t=66eef212-08d6-5e6c-a560-7d01149d0188&l=7812) 
```
CMake Error at scripts/cmake/vcpkg_replace_string.cmake:17 (message):
  vcpkg_replace_string made no changes.
Call Stack (most recent call first):
  ports/fmt/portfile.cmake:25 (vcpkg_replace_string)
  scripts/ports.cmake:192 (include)
```

File: ports/fmt/portfile.cmake
Comment (WangWeiLin-MV) [inline ports/fmt/portfile.cmake:1]:
```diff
-if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
-    vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/include/fmt/core.h
-        "defined(FMT_SHARED)"
-        "1"
-    )
-endif()
```

CMake config have already contains the definition `FMT_SHARED`

```cmake
set_target_properties(fmt::fmt PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "FMT_SHARED"
  INTERFACE_COMPILE_FEATURES "cxx_std_11"
  INTERFACE_COMPILE_OPTIONS "\$<\$<COMPILE_LANGUAGE:CXX>:/utf-8>"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
)
```

Comment (dg0yt) [reply ports/fmt/portfile.cmake:1]: @WangWeiLin-MV CMake config doesn't help msbuild users. That's why many ports carry the header fixup. D

**Comment by dg0yt** [comment]
#39738 seems to be a step ahead.

**Comment by waywardmonkeys** [comment]
Letting the other PR run ...

**Pull request closed without merging**